jl4/src/L4/TypeCheck.hs

```diff
@@ -582,10 +652,38 @@ addError e = do
   ctx <- use #errorContext
   with (MkCheckErrorWithContext e ctx)
 
+-- | Should never return 'Nothing' if our system is OK.
+getEntityInfo :: Resolved -> Check (Maybe CheckEntity)
+getEntityInfo r = do
+  ei <- use #entityInfo
+  pure (snd <$> Map.lookup (getUnique r) ei)
```

Comment:
why not return `Check CheckEntity` then - I have seen this used in another place where this would return `Nothing` for that function if this were a `Nothing`, I think I'd rather have this explode here if we had a bug than silently keep on going and potentially returning wrong results from functions that use this?

Reply:
I'm really reluctant to introduce places where the server explodes as long as that makes the editor / language server integration crash weirdly. I'd be happy to throw an internal error *and* continue, but in order to continue, I'd still have to produce `Nothing` then, because there's no good default value.

Reply:
Added an internal error in 961f98c645ec4e37c344a1178c45704515291563.